packages/web-components/src/globals/mixins/csp-compliance.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { state } from 'lit-element';
+import { Constructor } from '../defs';
+
+/**
+ * @param Base The base class.
+ * @returns A mix-in that sets its defined stable selector.
+ */
+const CspComplianceMixin = <T extends Constructor<HTMLElement>>(Base: T) => {
+  class CspComplianceMixinImpl extends Base {
+    editableStyle?: HTMLStyleElement;
+
+    @state()
+    nonce?;
+
+    connectedCallback() {
+      // TS seems to miss Element function definitions
+      // @ts-ignore
+      super.connectedCallback();
+
+      // TS doesn't like custom window properties.
+      // @ts-ignore
+      const globalNonce = window.nonce;
+
+      if (globalNonce) {
+        this.nonce = globalNonce;
+      }
+
+      let editableStyle = document.querySelector(`head style.csp-compliance-styles`) as HTMLStyleElement;
+
+      if (!editableStyle) {
+        const newStyle = document.createElement('style');
+        newStyle.classList.add('csp-compliance-styles');
+        newStyle.innerHTML = `
+          :root {
+            --color: darksalmon;
+            background: var(--color);
+          }
+
+          body {
+            padding: 1rem;
+          }
```

Comment:
I think these might be leftover styles we could remove? Unless they have a purpose I'm not seeing 🤔

Reply:
Oops